src/detect_fig.py

```diff
@@ -0,0 +1,203 @@
+
+    Raises:
+        FileNotFoundError: ファイルが見つからない場合に発生
+    """
+    try:
+        if not os.path.exists(path):
+            raise FileNotFoundError(f"'{path}' does not found\n")
+
+    except FileNotFoundError as e:
+        print("\nError:", e)
+
+
+class Detect():
+    """yolov5(物体検出)をロボコン用に編集したクラス."""
+
+    DEVICE = 'cpu'
+    IMG_SIZE = (640, 480)
+
+    def __init__(self,
+                 img_path=IMAGE_PATH / 'test_image.png',
+                 weights=YOLO_PATH / 'learned_weight.pt',
+                 label_data=YOLO_PATH / 'label_data.yaml',
+                 conf_thres=0.6,
+                 iou_thres=0.45,
+                 max_det=10,
+                 line_thickness=3,
+                 stride=32):
+        """コンストラクタ.
+
+        Args:
+            img_path (str): 画像パス
+            weights (str): 重みファイルパス
+            label_data (str): ラベルを記述したファイルパス
+            conf_thres (float): 信頼度閾値
+            iou_thres (float): NMS IOU 閾値
+            max_det (int): 最大検出数
+            line_thickness (int): カメラID
+            stride (int): ストライド
+        """
+        check_exists(img_path)
+        check_exists(weights)
+        check_exists(label_data)
+        self.img_path = str(img_path)
+        self.weights = str(weights)
+        self.label_data = str(label_data)
+        self.conf_thres = conf_thres
+        self.iou_thres = iou_thres
+        self.max_det = max_det
+        self.line_thickness = line_thickness
+        self.stride = stride
+
+    def read_image(self) -> np.ndarray:
+        """画像を読み込む関数.
+
+        Args:
+            img_path: 画像パス
+        Returns:
+            im(np.ndarray): パディング処理を行った入力画像
+            im0(np.ndarray): 入力画像
+        """
+        im0 = cv2.imread(self.img_path)  # BGR
+
+        # リサイズとパディング処理
+        im = letterbox(im0, Detect.IMG_SIZE, stride=self.stride,
+                       auto=True)[0]
+        # BGR -> RGB
+        im = im.transpose((2, 0, 1))[::-1]
+        # 連続したメモリ領域に変換
+        im = np.ascontiguousarray(im)
+
+        return im, im0
+
+    def detect(self, save_path=None) -> list:
+        """物体の検出を行う関数.
+
+        Args:
+            save_path(str): 検出結果の画像保存パス
+                            Noneの場合、保存しない
+        Returns:
+            pred: 予測結果
+        """
+        # cpuを指定
+        device = select_device(Detect.DEVICE)
+
+        # モデルの読み込み
+        model = DetectMultiBackend(self.weights,
+                                   device=device,
+                                   dnn=False,
+                                   data=self.label_data,
+                                   fp16=False)
+
+        stride, labels, pt = model.stride, model.names, model.pt
+
+        # 画像のサイズを指定されたストライド（ステップ）の倍数に合わせるための関数
+        img_size = check_img_size(Detect.IMG_SIZE, s=stride)  # >> [640, 640]
+
+        # モデルの初期化
+        batch_size = 1
+        model.warmup(
+            imgsz=(1 if pt or model.triton else batch_size, 3, *img_size))
+
+        # 画像の読み込み
```

Comment:
大量に使うからimって短い名前にしてると思うけど、せめて、なんなのかのコメントが欲しい。

```suggestion
        # 画像の読み込み im:バウンディング処理画像, im0:入力画像
```